### PR TITLE
Enable pre-commit checks

### LINKED ISSUE #60: Enable pre-commit checks
Here is the list of checks I propose to add:
- [x] Pre-commit hooks (they go directly under [pre-commit-hooks](https://github.com/pre-commit/pre-commit-hooks) repo).
  - [x] `end-of-file-fixer` adds new line at the end of file.
  - [x] `trailing-whitespace` remove trailing whitespaces.
  - [x] `check-added-large-files` - prevents giant files from being committed.
- [x] [flake8](https://github.com/pycqa/flake8). Additional flake8-plugins:
  - [x] [flake8-bugbear](https://pypi.org/project/flake8-bugbear/) for detecting bugs and potential design problems.
  - [x] [flake8-builtins](https://pypi.org/project/flake8-builtins/) for verifying that builtins are not used as variable names.
  - [x] [flake8-comprehensions](https://pypi.org/project/flake8-comprehensions/) for writing better and consistent comprehensions.
  - [x] [flake8-debugger](https://pypi.org/project/flake8-debugger/) for checking that there are no forgotten [breakpoints](https://docs.python.org/3/library/functions.html#breakpoint).
  - [x] ~[flake8-eradicate](https://pypi.org/project/flake8-eradicate/) to avoid “dead” or commented code~ (❌  couldn't be installed due to the dependency conflict).
  - [x] [flake8-logging-format](https://github.com/globality-corp/flake8-logging-format) for consistent logging.
  - [x] [pep8-naming](https://pypi.org/project/pep8-naming/) for verifying that PEP8 naming conventions are followed.
  - [x] [pyflakes](https://github.com/PyCQA/pyflakes) for checking Python source files for various errors.
  - [x] [tryceratops](https://pypi.org/project/tryceratops/) to avoid exception anti-patterns.
- [ ] Type checking: [mypy](https://github.com/python/mypy).
- [x] [isort](https://pycqa.github.io/isort/) for sorting imports.
- [x] [autoflake](https://github.com/PyCQA/autoflake) for removing unused imports and variable definitions.
- [ ] [black](https://github.com/psf/black) for formatting the rest of the code.
- [x] [pyupgrade](https://github.com/asottile/pyupgrade) to automatically upgrade syntax for newer versions of the language.
- [x] ~[yamllint](https://github.com/adrienverge/yamllint)~ (❌  conflicts with yamlfmt, even though this combination is [recommended officially](https://github.com/jumanjihouse/pre-commit-hook-yamlfmt#combine-with-yamllint)) and [yamlfmt](https://github.com/jumanjihouse/pre-commit-hook-yamlfmt) for YAML files formatting.

**important**

To fix this issue, it is enough to agree on the list of checks we want to have and to add them. However, fixing the problems reported by those checks may but doesn't have to be included in the corresponding PR. I assume there will be a lot of those.

**checked items in the list** correspond to the checks I've already implemented in #62 

**full list of hooks** can be found [here](https://pre-commit.com/hooks.html)

### PR DESCRIPTION
**Pull request by yakutovicha**
fixes #60

### CODE REVIEW

**Comment by yakutovicha** [comment]
For the moment I added checks that seem to be good in general. Please let me know if you are fine with my choice.

**Review by NikoOinonen — COMMENTED** [review]
Looks ok to me except for the point on automatic removal of function arguments.

File: dev/bakup/GUImorse.py [hunk 1]
```diff
@@ -265,28 +265,28 @@ def updateFF(self):
...
     def updateFromFF(self):
         self.updateFF()
         self.getFF()
         self.upload_and_relax()
 
     def loadInputs_New(self):
-        print("self.mode  ", self.mode) 
+        print("self.mode  ", self.mode)
         try:
-            with open('atomtypes.ini', 'r') as f:  self.str_Species = f.read();
```
[the remark this answers is not in the document]
Comment (yakutovicha) [reply dev/bakup/GUImorse.py:278]: This is done by pyupgrade, and (according to my understanding) it has some [special treatment of the `open` function](https://github.com/asottile/pyupgrade#redundant-open-modes). Other functions that use the default value of an argument are not affected 👍 .

**Comment by yakutovicha** [comment]
I would like to request **explicit** approval from @ondrejkrejci and @ProkopHapala as well. This PR will influence the code styling and will enforce users to adhere to it in the future. So if you want to say no to some specific check, this would be the moment 😎 .

File: dev/bakup/GUImorse.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (NikoOinonen) [reply dev/bakup/GUImorse.py:278]: Ok, that's fine for me then.

**Comment by yakutovicha** [comment]
@ondrejkrejci, @ProkopHapala since you didn't provide your opinion yet, should I consider that PR good to go? Or do you have issues with it?

**Comment by ondrejkrejci** [comment]
I put the comment there, that I wanted to suggest maybe some changes. I am saturated now. **Friday** - So if you do not hear from me on Monday, then feel free to go. Sorry for the delay. I am trying to sit on too many chairs now, so this had to go to very low priority for me ....

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+---
+repos:
+    - repo: https://github.com/pre-commit/pre-commit-hooks
+      rev: v4.4.0
+      hooks:
+          - id: end-of-file-fixer
+          - id: trailing-whitespace
+          - id: check-yaml
+          - id: check-added-large-files
```
Comment (ProkopHapala) [inline .pre-commit-config.yaml:9]: great this is useful!

**Review by ProkopHapala — APPROVED** [review]

**Comment by yakutovicha** [comment]
Since everyone agreed to keep supporting Python 3.6+, I will redo some of the changes and re-request everybody's approval.

**Comment by yakutovicha** [comment]
Since the only change was enabling Python 3.6+ compatibility, I decided to go without a further review. All the other changes were already approved.

File: ppafm/HighLevel.py [hunk 3]
```diff
@@ -54,13 +54,13 @@ def relaxedScan3D( xTips, yTips, zTips, trj=None, bF3d=False ):
...
         sys.stdout.flush()
-        sys.stdout.write("\rrelax ix: {}".format(ix))
+        sys.stdout.write(f"\rrelax ix: {ix}")
         sys.stdout.flush()
         for iy,y in enumerate( yTips  ):
             rTips[:,0] = trj[:,0] + x
             rTips[:,1] = trj[:,1] + y
             rTips[:,2] = trj[:,2]
-            itrav = core.relaxTipStroke( rTips, rs, fs ) / float( len(zTips) )
+            core.relaxTipStroke( rTips, rs, fs ) / float( len(zTips) )
```
Comment (ondrejkrejci) [inline ppafm/HighLevel.py:63]: Hey, really, really sorry for being slow (as I have told, this has lower priority than my immediate work, with 2 new students, that I am advising ...) - we spoke about this particular line yesterday with @ProkopHapala - from my point of view, the `/ float( len(zTips) )` is not necessary at all.

File: ppafm/chemistry.py [hunk 4]
```diff
@@ -194,7 +195,7 @@ def ringsToMolecule( ring_pos, ring_Rs, Lrange=6.0 ):
 def normalizeSpeciesProbs( species ):
     out = []
     for l in species:
-        renorm=1.0/sum( s[1] for s in l )
+        1.0/sum( s[1] for s in l )
```
Comment (ondrejkrejci) [inline ppafm/chemistry.py:198]: Does this whole procedure even do something - I do not see any adding to the `out` list in here ...

File: ppafm/common.py [hunk 5]
```diff
@@ -87,13 +87,13 @@ def Fz2df( F, dz=0.1, k0 = params['kCantilever'], f0=params['f0Cantilever'], n=4
...
     '''
-    conversion of vertical force Fz to frequency shift 
+    conversion of vertical force Fz to frequency shift
     according to:
     Giessibl, F. J. A direct method to calculate tip-sample forces from frequency shifts in frequency-modulation atomic force microscopy Appl. Phys. Lett. 78, 123 (2001)
     oscialltion amplitude of cantilever is A = n * dz
     '''
     dr = np.sqrt( d[0]**2 + d[1]**2 + d[2]**2 )
-    hat=np.array(d)/dr
+    np.array(d)/dr
```
Comment (ondrejkrejci) [inline ppafm/common.py:96]: Once more, does this even do something here.

File: ppafm/ocl/AFMulator.py [hunk 6]
```diff
@@ -457,7 +457,6 @@ def quick_afm(file_path, scan_size=(16, 16), offset=(0, 0), distance=8.0, scan_s
             QZs = [0, 0, 0, 0]
         elif tip == 'pz':
             Qs = [10*charge, -10*charge, 0, 0]
-            Qzs = [0.1, -0.1, 0, 0]
```
Comment (ondrejkrejci) [inline ppafm/ocl/AFMulator.py:460]: I am not quite sure, if you wast to remove this line @NikoOinonen - could you look at it?

Comment (NikoOinonen) [reply ppafm/ocl/AFMulator.py:460]: I missed this before. Yeah, this does not look right. Why would it remove this line?

File: ppafm/chemistry.py (hunk 4 above)
Comment (ProkopHapala) [reply ppafm/chemistry.py:198]: probably not, it seems not to be used anywhere, it is perhaps just remnant of development process

File: ppafm/common.py (hunk 5 above)
Comment (ProkopHapala) [reply ppafm/common.py:96]: the line does nothing, but the whole function does.
Perhaps I was originaly thingkin I need normalized vectro, but then realized I don't.
Yet I'm not sure about that now. maybe there should be 

return (dFconv_x*hat[0] + dFconv_y*hat[1] + dFconv_z*hat[2] )*units*f0/k0

File: ppafm/ocl/AFMulator.py (hunk 6 above)
Comment (ProkopHapala) [reply ppafm/ocl/AFMulator.py:460]: yes, I also think this is wrong (should not be removed)

**Comment by yakutovicha** [comment]
@ondrejkrejci, can you please make a follow-up PR to fix the issues you've reported?

**Pull request merged**